doc/faq.rst

```diff
@@ -122,3 +136,125 @@ file:
 
 The ``ocaml-print-intf`` program has special support for Dune, so it will
 automatically understand external dependencies.
+
+How Can I Build a Single Library?
+=================================
+
+You might want to do this when you don't have all the dependencies installed to
+compile an entire project, or parts of the project don't build for whatever
+reason. Maybe you want to check if your changes compile or produce build
+artifacts needed by ``ocaml-lsp-server``.
+
+Suppose you have a library defined in ``src/foo/dune``:
+
+.. code:: dune
+
+  (library
+   (public_name my_library)
+   ...)
+
+You can build this library on its own by running the following from the project
+root directory:
+
+.. code:: bash
+
+   $ dune build %{cmxa:src/foo/my_library}
+
+Note that the path (``src/foo`` in the example above) is relative to the current
+directory - not the project root. If the library defines a ``name`` distinct from
+its ``public_name`` then that can be used interchangeably with the ``public_name``
+in this command.
+
+Files and Directories Whose Names Begin with "." (Period) are Ignored by ``source_tree``
+======================================================================================
```

Comment:
Title underline is too short (see the docs CI)

Reply:
oops, updated the title without updating the underline. fixed